### PR TITLE
Suggest `where` keyword in generic local function declarations

### LINKED ISSUE #72821: `where` is not suggested when adding the first type constraint to a local function
## Version Used
VS 2022 Version 17.9.5
Roslyn 4.9.0 (a98c90d5645537)

## Steps to Reproduce

1. Begin declaring a generic local function with at least one type parameter
2. After closing the argument list parentheses, begin typing `where`

Sample code:
```csharp
static void Outer()
{
    int Inner<T, U, V>()
        // place cursor here
    {
    }
}
```

## Expected Behavior

`where` is suggested, given the presence of type parameter in the local function.

## Actual Behavior

The `where` keyword is not suggested at all. In fact, nothing is suggested in that area.

### Note

After adding the first `where` clause, typing `where` to initiate another clause properly suggests the `where` keyword.
For example, in the following code:
```csharp
static void Outer()
{
    int Inner<T, U, V>()
        where T : class
        // place cursor here
    {
    }
}
```
the `where` keyword is suggested.

### PR DESCRIPTION
**Pull request by Rekkonnect**
Closes #72821

### CODE REVIEW

**Comment by DoctorKrolic** [comment]
> This also adds a public API:

Public APIs need to go through API review process first. You cannot just add public API to simplify your IDE fix as you go

**Comment by Rekkonnect** [comment]
@sharwell removed the public API, ptal

**Review by sharwell — APPROVED** [review]

File: src/EditorFeatures/CSharpTest2/Recommendations/WhereKeywordRecommenderTests.cs [hunk 1]
```diff
@@ -631,5 +631,46 @@ public class C<T> where T : List<(int, string)> $$
                 }
                 """);
         }
+
+        [Fact]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest2/Recommendations/WhereKeywordRecommenderTests.cs:635]:
```suggestion
        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72821")]
```

File: src/EditorFeatures/CSharpTest2/Recommendations/WhereKeywordRecommenderTests.cs [hunk 2]
```diff
@@ -631,5 +631,46 @@ public class C<T> where T : List<(int, string)> $$
...
+                """
+                class C
+                {
+                    void M<T>()
+                    {
+                        void Inner() $$
+                """);
+        }
+
+        [Fact]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest2/Recommendations/WhereKeywordRecommenderTests.cs:648]:
```suggestion
        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72821")]
```

File: src/EditorFeatures/CSharpTest2/Recommendations/WhereKeywordRecommenderTests.cs [hunk 3]
```diff
@@ -631,5 +631,46 @@ public class C<T> where T : List<(int, string)> $$
...
+                """
+                class C
+                {
+                    void M<T>()
+                    {
+                        void Inner<T1>() $$
+                """);
+        }
+
+        [Fact]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest2/Recommendations/WhereKeywordRecommenderTests.cs:661]:
```suggestion
        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72821")]
```

**Review by CyrusNajmabadi — APPROVED** [review]
lgtm with suggestions.

**Pull request merged**